// eslint-config-rw/tailwind.config.js
const colors = require('tailwindcss/colors');
module.exports = {
  darkMode: ['class','[data-theme="dark"]'],
  content:[
    '../../packages/reactwiind/src/**/*.{js,ts,tsx,mdx}',
    './src/**/*.{js,ts,tsx,mdx}',
    './stories/**/*.{js,ts,tsx,mdx}',
    './pages/**/*.{js,ts,jsx,tsx,mdx}',
    './apps/**/*.{js,ts,jsx,tsx,mdx}'
  ],
//new
  // content: [
  //   '../../packages/reactwiind/src/**/*.{ts,tsx}',
  //   './src/**/*.{ts,tsx}',
  //   './stories/**/*.{ts,tsx,mdx}',
  //   './pages/**/*.{js,ts,jsx,tsx}',
  //   './apps/**/*.{js,ts,jsx,tsx}'
  // ],
  theme: {
    colors: {
      primary:{
        light: "#172941",
        DEFAULT:"#081D3B",
        dark: "#0E1927",
      },
      onprimary:{
        DEFAULT:"#fff",
      },
      secondary: {
        light: "#3171DE",
        DEFAULT:"#3072E0",
        dark: "#2458AC",
      },
      onsecondary:{
        DEFAULT:"#fff",
      },
      background: {
        DEFAULT:"#f2f2f2",
      },
      onbackground: {
        DEFAULT:"#1E1E1D",
      },
      surface: {
          light: "#f2f2f2",
          DEFAULT:"#081D3B",
          dark: "#C1C1C1",
      },
      onsurface: {
        DEFAULT:"#1E1E1D",
      },
      success: {
        DEFAULT:"#22bb33",
      },
      warning: {
        DEFAULT:"#f0ad4e",
      },
      danger: {
        DEFAULT:"#bb2124",
      },
    ...colors
  },
  fontFamily:{
    sans: ['Graphik', 'sans-serif'],
    serif: ['Merriweather', 'serif']
  }

}
}
