// reactwiind-pro/src/FeaturedImageList/index.tsx
import type { AppProps } from 'next/app';
import {HeroImage, Button, useTheme, Typography, FeaturedImage} from 'reactwiind';
import {Adult, NewBorn,OneMonth, NineMonths, TwoYears} from "../svg/brainconnects";
import {BoyGirlWorld} from "../svg/BoyGirlWorld";
import { FC } from 'react';

 

export interface WingoFeaturedImageListItemProp {
    Icon: JSX.Element;
    name: string;
    descname: string;
    title: string;
    desc: string;
}

export interface WingoFeaturedImageListProp {
    list?: Array<WingoFeaturedImageListItemProp>;
    title?: React.ReactNode | string;
}

// export function WingoProgram({icon:Icon,name,descname,title, desc}: WingoProgramProp) {
//   return <div className="grid grid-cols-2">
//      <div className=""><Icon /></div>
//      <div className="flex flex-col">
//        <Typography   variant="h4">{title}</Typography>
//        <Typography   variant="paragraph">{desc}</Typography>
//     </div>
//   </div>
// }

// const programs = [
   
//     {
//         heading:"Wingo Tots Program",
//         name: "Wingo Tots",
//         subheading:"1.8 - 2.5 years",
//         desc:"Our Wingo junior/Nursery program is designed to develop reading, writing, social, logical, rhyming, other important skills. Children are involved in creative, innovative activities to create a learning experience that help create brain connections, hence develop a strong foundation of our children. We are committed towards e-learning to make learning more easy, available and in a fun way. Our digital studio class ensure we take advantage of technology to create a creative, innovative learning experience and faster development.",
//         Icon: BoyGirlWorld
//     },
//     {
//         name: "Wingo Junior",
//         subheading:"2.5 - 3.5 years",
//         heading:"Wingo Junior Program",
//         desc:"Our Wingo junior/Nursery program is designed to develop reading, writing, social, logical, rhyming, other important skills. Children are involved in creative, innovative activities to create a learning experience that help create brain connections, hence develop a strong foundation of our children. We are committed towards e-learning to make learning more easy, available and in a fun way. Our digital studio class ensure we take advantage of technology to create a creative, innovative learning experience and faster development.",
//         Icon: BoyGirlWorld,
//         swapcols: true
//     },
//     {
//         name: "Wingo Senior",
//         subheading:"3.5 - 5 years",
//         heading:"Wingo Senior Program",
//         desc:"Our Wingo junior/Nursery program is designed to develop reading, writing, social, logical, rhyming, other important skills. Children are involved in creative, innovative activities to create a learning experience that help create brain connections, hence develop a strong foundation of our children. We are committed towards e-learning to make learning more easy, available and in a fun way. Our digital studio class ensure we take advantage of technology to create a creative, innovative learning experience and faster development.",
//         Icon: BoyGirlWorld
//     }
// ]

export const  WingoFeaturedImageList:FC<WingoFeaturedImageListProp> = ({title, list}) => {
    const {getStyles} = useTheme();
  return <div data-featured-image-list-component>  
     
      { (typeof title == 'string')?<Typography   variant="h2" className="mb-6">{title}</Typography>:title}
   
     <div data-featured-image-list>
        {
            list?.map((content:WingoFeaturedImageListItemProp)=>{
                return <FeaturedImage {...content}/>
            })
        }
    </div>
  </div>  
}

WingoFeaturedImageList.defaultProps = {
    title: "Feature Heading",
    list: []
}
