// eslint-config-rw/config.js
const colors = require('tailwindcss/colors');

const merge = require("deepmerge");
 
const config = {
  darkMode: 'class',
  content:[
    "./node_modules/reactwiind/**/*.{js,ts,jsx,tsx,mdx}",
    '../../packages/reactwiind/src/**/*.{js,ts,tsx,mdx}',
    './src/**/*.{js,ts,tsx,mdx}',
    './stories/**/*.{js,ts,tsx,mdx}',
    './pages/**/*.{js,ts,jsx,tsx,mdx}',
    './apps/**/*.{js,ts,jsx,tsx,mdx}'
  ],
  theme: {
      colors: {
        primary: {
          light: "#2b2d42",
          DEFAULT:"#2b2d42",  
          dark: "#2b2d42",
        },
        onprimary: {
          DEFAULT:"#edf2f4",
        },
        secondary: {
          light: "#8d99ae",
          DEFAULT:"#8d99ae",
          dark: "#8d99ae",
        },
        onsecondary: {
          DEFAULT:"#d90429",
        },
        background: {
          DEFAULT:"#f2f2f2",
        },
        onbackground: {
          DEFAULT:"#d90429",
        },
        surface: {
            light: "#2b2d42",
            DEFAULT:"#2b2d42",
            dark: "#2b2d42",
        },
        onsurface: {
          DEFAULT:"#d90429",
        },
        success: {
          DEFAULT:"#22bb33",
        },
        warning: {
          DEFAULT:"#f0ad4e",
        },
        danger: {
          DEFAULT:"#bb2124",
        },
        ...colors

    }
  },
  fontFamily: {
    sans: ['Graphik', 'sans-serif'],
    serif: ['Merriweather', 'serif']
  }
}

function withReactwiind(tconfig) {
  const mergedConfig = merge(config, {...tconfig});
  console.log("-----------------");
  console.log("-----------------");
  console.log("-----------------");
  console.log("-----------------");
  console.log("merged config-",JSON.stringify(mergedConfig))
  return mergedConfig;
}


module.exports =  withReactwiind;
