// reactwiind-pro/src/Lists/ImageTextList.tsx
import {useTheme, Typography, ImageText, type ImageTextProps} from 'reactwiind';
import {Adult, NewBorn,OneMonth, NineMonths, TwoYears} from "rw-assets";


import { FC } from 'react';

 
 
export interface ImageTextListProps {
    list?: Array<ImageTextProps>;
    heading?: React.ReactNode | string;
}

const defaultProps = {
    heading: "brain progression for kids on different stages",
    list: [
        {
            img:NewBorn,
            vertical: true,
            caption: "New Born Baby"
        },
        {
            img:OneMonth,
            vertical: true,
            caption: "1 Month Old"
        }, 
        {
            img:NineMonths,
            vertical: true,
            caption: "9 Months"
        }, 
        {
            img:TwoYears,
            vertical: true,
            caption: "2 years"
        },
        {
            img:Adult,
            vertical: true,
            caption: "Adult"
        }
    ]
}

export const  ImageTextList:FC<ImageTextListProps> = ({heading, list=defaultProps.list}) => {
    const {getStyles} = useTheme();
  return <div data-featured-image-list-component>  
     
      { (typeof heading == 'string')?<Typography   variant="h2" className="mb-6">{heading}</Typography>:heading}
   
     <div data-Images-Text-list className="flex justify-around">
        {
            list?.map((imageTextItem:ImageTextProps)=>{
                return <ImageText {...imageTextItem}/>
            })
        }
    </div>
  </div>  
}
