// reactwiind-pro/src/Lists/ImageContentList.tsx
import {Hero, Button, useTheme, Typography, ImageContent, type ImageContentProps} from 'reactwiind';
import {Adult, NewBorn,OneMonth, NineMonths, TwoYears} from "rw-assets";
import {BoyGirlWorld, BabyRainbowSlide, SuperKids } from "rw-assets";
import { FC } from 'react';

export interface ImageContentListProps {
    list?: Array<ImageContentProps>;
    heading?: React.ReactNode | string;
}
 
const defaultProps:ImageContentListProps = {
    heading: "Programs we offer",
    list:[
            {
                heading:"Wingo Tots Program",
                subheading:"1.8 - 2.5 years",
                desc:"Our Wingo junior/Nursery program is designed to develop reading, writing, social, logical, rhyming, other important skills. Children are involved in creative, innovative activities to create a learning experience that help create brain connections, hence develop a strong foundation of our children. We are committed towards e-learning to make learning more easy, available and in a fun way. Our digital studio class ensure we take advantage of technology to create a creative, innovative learning experience and faster development.",
                img: BoyGirlWorld,
                swapcols: false
            },
            {
              
                subheading:"2.5 - 3.5 years",
                heading:"Wingo Junior Program",
                desc:"Our Wingo junior/Nursery program is designed to develop reading, writing, social, logical, rhyming, other important skills. Children are involved in creative, innovative activities to create a learning experience that help create brain connections, hence develop a strong foundation of our children. We are committed towards e-learning to make learning more easy, available and in a fun way. Our digital studio class ensure we take advantage of technology to create a creative, innovative learning experience and faster development.",
                img: BabyRainbowSlide,
                swapcols: true 
            },
            {
                subheading:"3.5 - 5 years",
                heading:"Wingo Senior Program",
                desc:"Our Wingo junior/Nursery program is designed to develop reading, writing, social, logical, rhyming, other important skills. Children are involved in creative, innovative activities to create a learning experience that help create brain connections, hence develop a strong foundation of our children. We are committed towards e-learning to make learning more easy, available and in a fun way. Our digital studio class ensure we take advantage of technology to create a creative, innovative learning experience and faster development.",
                img: SuperKids,
                swapcols: false
            }
        ] as Array<ImageContentProps>
}

export const  ImageContentList:FC<ImageContentListProps> = ({heading=defaultProps.heading, list = defaultProps.list}) => {
    const {getStyles} = useTheme();
  return <div data-featured-image-list-component>  
     
      { (typeof heading == 'string')?<Typography   variant="h2" className="mb-6">{heading}</Typography>:heading}
   
     <div data-featured-image-content-list>
        {
            list?.map((content:ImageContentProps)=>{
                return <ImageContent {...content}/>
            })
        }
    </div>
  </div>  
}
