// reactwiind-pro/src/Footer/WingoFooter.tsx
import {NavBar, Footer, LinkProps, Spacer} from 'reactwiind';
import { ComponentProps, FC, ReactNode } from 'react';
 import {Social} from "./Social";
 import {MindWingoYellow} from "rw-assets";
 import {Links} from "../Lists";
import {Icon} from "rw-assets"
 

export interface WingoFooterCol {
   logo?: ReactNode;
   social?: ReactNode;
   heading?: string;
   heading2?: string;
   list?: Array<LinkProps>;
   list2?: Array<LinkProps>;
}


export interface WingoFooterProps {
  // menuitems?: Array<WingoHeaderMenuItem>;
  // link?: string;
  // logo?: FC<ComponentProps<'svg'>> | null;
  col1:WingoFooterCol;
  col2: WingoFooterCol;
  col3: WingoFooterCol;
  bgcolor?:string;
}

const timing:Array<LinkProps> = []


export const  WingoFooter:FC<WingoFooterProps> = ({bgcolor="bg-secondary", col1, col2, col3}) => {
  const {logo=<MindWingoYellow />, list:listCol1, heading:headingCol1} = col1;
  const {list:listCol2, list2:list2Col2, heading:headingList1Col2, heading2:heading2List2Col2} = col2;
  const {social, list: listCol3, heading:headingCol3} = col3;
 

  return  <Footer>
            <Footer.Column>
              <div className="flex flex-col">
                {logo}
                <Links heading={headingCol1} links={listCol1}/>
               </div>
            </Footer.Column>

            <Footer.Column>
              <div className="flex flex-col">
                <Spacer className="py-8"/>
                <Links heading={headingList1Col2} links={listCol2}/>
                <Spacer className="py-1"/>
                <Links heading={heading2List2Col2} links={list2Col2}/>
               
               </div>
            </Footer.Column>
            
            <Footer.Column>
                 <div className="flex flex-col">
                     {social}
                     <Spacer className="py-1"/>
                    <Links heading={headingCol3} links={listCol3}/>
                  </div>
            </Footer.Column>
          </Footer>
}
